animation-glib/transform/transform.cpp

```diff
@@ -0,0 +1,202 @@
+}
+
+float
+animation_transform_animation_progress (AnimationTransformAnimation *transform_animation)
+{
+  AnimationTransformAnimationPrivate *priv =
+    reinterpret_cast <AnimationTransformAnimationPrivate *> (animation_transform_animation_get_instance_private (transform_animation));
+
+  return priv->interface->Progress ();
+}
+
+static void
+animation_transform_animation_set_property (GObject      *object,
+                                            guint         prop_id,
+                                            const GValue *value,
+                                            GParamSpec   *pspec)
+{
+  AnimationTransformAnimation *transform_animation = ANIMATION_TRANSFORM_ANIMATION (object);
+  AnimationTransformAnimationPrivate *priv =
+    reinterpret_cast <AnimationTransformAnimationPrivate *> (animation_transform_animation_get_instance_private (transform_animation));
+
+  switch (prop_id)
+    {
+    case PROP_INTERFACE:
+      priv->interface = reinterpret_cast <at::TransformAnimation *> (g_value_get_pointer (value));
+      break;
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+    }
+}
+
+static void
+animation_transform_animation_get_property (GObject    *object,
+                                            guint       prop_id,
+                                            GValue     *value,
+                                            GParamSpec *pspec)
+{
+  AnimationTransformAnimation *transform_animation = ANIMATION_TRANSFORM_ANIMATION (object);
+  AnimationTransformAnimationPrivate *priv =
+    reinterpret_cast <AnimationTransformAnimationPrivate *> (animation_transform_animation_get_instance_private (transform_animation));
+
+  switch (prop_id)
+    {
+    case PROP_INTERFACE:
+      g_value_set_pointer (value, reinterpret_cast <gpointer> (priv->interface));
+      break;
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+    }
+}
+
+static void
+animation_transform_animation_finalize (GObject *object)
+{
+  AnimationTransformAnimation *transform_animation = ANIMATION_TRANSFORM_ANIMATION (object);
+  AnimationTransformAnimationPrivate *priv =
+    reinterpret_cast <AnimationTransformAnimationPrivate *> (animation_transform_animation_get_instance_private (transform_animation));
+
+  delete priv->interface;
+  priv->interface = nullptr;
+
+  G_OBJECT_CLASS (animation_transform_animation_parent_class)->finalize (object);
+}
+
+static void
+animation_transform_animation_init (AnimationTransformAnimation *model)
+{
+}
+
+
+static void
+animation_transform_animation_class_init (AnimationTransformAnimationClass *klass)
+{
+  GObjectClass *object_class = G_OBJECT_CLASS (klass);
+
+  object_class->get_property = animation_transform_animation_get_property;
+  object_class->set_property = animation_transform_animation_set_property;
+  object_class->finalize = animation_transform_animation_finalize;
+
+  animation_transform_animation_props[PROP_INTERFACE] =
```

Comment:
It is. Actually, it may be easier to not expose it through gobject at all and just keep it as an internal member that gets set at the constructor? That would probably simplify a few things..m